[상품] 미발행된 이벤트를 발행하는 스케줄러 구현

qnmgjeklnbrl opened the pull request:
[OutBoxScheduler]
- Outbox 테이블에서 미발행(READY_TO_PUBLISH) 상태의 이벤트를 주기적으로 조회하여 Kafka에 발행
- 실행간격은 100ms부터 백오프 방식으로 최대 10분까지 늘어남

로컬 db에 테스트 데이터 넣어서 동작 테스트했습니다.

먼저 잘 개발해주셔서 참고 많이 했습니다 ㅎ.ㅎ

qnmgjeklnbrl committed:
chore: 당첨 이벤트 레코드 이름 변경

- DecreaseStockEvent.java -> DrawWinEvent.java로 클래스명 변경

qnmgjeklnbrl committed:
chore: 카프카 토픽명 수정

- prize.decrease -> draw.win
- decrease.result -> prize.decrease_result

qnmgjeklnbrl committed:
feat: outbox 스케줄러 구현

dlehddn reviewed (changes requested):
고생하셨습니다:)
수정이 필요한 부분 및 개인적인 피드백 코멘트로 남겼으니 확인해주세요.

qnmgjeklnbrl committed:
refactor: dto 패키지 구조 변경, ExternalEvent 인터페이스 도입

- dto 하위 패키지를 목적별로 분할
- ExternalEvent 인터페이스 도입
- 응모 식별을 위해 이벤트 객체들에 drawId필드 추가
- 이벤트 발행 시점 식별을 위해 occurAt 필드 추가

dlehddn reviewed (approved):
고생하셨습니다ㅎ
